internal/packages/installer/factory.go

```diff
@@ -54,11 +57,20 @@ func NewForPackage(options Options) (Installer, error) {
 		return nil, fmt.Errorf("failed to get kibana version: %w", err)
 	}
 
-	supportsZip := !version.LessThan(semver8_7_0)
+	supportsUploadZip, err := isAllowedInstallationViaApi(context.TODO(), options.Kibana, version)
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate whether or not it can be used upload API: %w", err)
+	}
 	if options.ZipPath != "" {
-		if !supportsZip {
-			return nil, fmt.Errorf("not supported uploading zip packages in Kibana %s (%s required)", version, semver8_7_0)
+		if !supportsUploadZip {
+			if version.LessThan(semver8_7_0) {
+				return nil, fmt.Errorf("not supported uploading zip packages in Kibana %s (%s required)", version, semver8_7_0)
+			}
+			if version.LessThan(semver8_8_2) {
+				return nil, fmt.Errorf("not supported uploading zip packages in Kibana %s (%s required or Enteprise license)", version, semver8_8_2)
+			}
```

Comment:
Nit. Maybe `isAllowedInstallationViaApi` could also return the reason, as we are checking with the same versions as here.

Reply:
Added a new return parameter that indicates the reason, just in the case there is an expected failure.

Reply:
Applied here https://github.com/elastic/elastic-package/pull/2511/commits/05e082b9311a22029c672ccb3af4365be22d7870